File: Backend/markitAPI/markit.Application/Features/Collections/Queries/ViewModels/CollectionItem.cs
namespace markit.Application.Features.Collections.Queries.ViewModels
{
    public class CollectionItem
    {
        public string Id { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public int CollectionId { get; set; }
        public CollectionItemType Type { get; set; }
        public int TypeId { get; set; }
        public string Preview {  get; set; } = string.Empty;
        public bool IsFavorite { get; set; }
        public string? Emoji { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }

    public enum CollectionItemType
    {
        Collection,
        Mark
    }
}

File: Frontend/src/app/workplace/interfaces/collection-item.ts
export interface CollectionItem {
    id: string;
    name: string;
    collectionId : number;
    type: CollectionItemType;
    typeId : number;
    isFavorite : boolean;
    preview?: string;
    createdAt?: Date;
    updatedAt?: Date;
    //* Utility properties
    updating?: boolean;
    emoji?: string;
}

export enum CollectionItemType {
    Collection = 'collection',
    Mark = 'mark'
}

export enum CollectionItemAction {
    Add = 'New',
    Rename = 'Rename',
    Delete = 'Delete'
}